spec/spec.md

```diff
@@ -0,0 +1,111 @@
+
+#### Domain Object
+
+A "domain object" is defined as an instance of a class from the application's code and not part of the SDK.
+
+## Versioning
+
+### Spec Versioning
+
+This spec follows [semantic versioning 2.0](https://semver.org/spec/v2.0.0.html). This spec also directly matches the Dapr runtime API version for major and minor. It means that for each minor version release of Dapr runtime, there will be a new release of the spec with the same version. The patch version of the spec and runtime increase independently based on fixes on each. For example: The spec version 1.9 is based on the runtime version 1.9, regardless of the patch version of each.
+
+The spec will be worked on in the main branch and include the `-alpha` suffix until it reached a stable version. Once the first stable version of the spec is reached, it will be under the release-x.y branch and tagged with vX.Y.0, such as the runtime release process.
+
+This allows SDK developers to claim compliance towards a fixed spec and runtime version instead of a moving target.
+
+### SDK Versioning
+
+An SDK MUST start as one repository in the [dapr-sandbox organization](https://github.com/dapr-sandbox) at version 0.1, following the process to onboard a new Dapr sandbox project. Before an SDK can graduate to the [dapr organization](https://github.com/dapr), it MUST comply to a stable version of this spec, in addition to the graduation criteria for Dapr sandbox projects.
+
+## Client Object
+
+The SDK MUST offer a way to instantiate a client object without requiring any parameter, using the `DAPR_HTTP_PORT` or `DAPR_GRPC_PORT` environment variables injected by the Dapr sidecar injector or Dapr CLI - defaulting to `3500` or `50001`, correspondingly, if environment variable is not set.
+
+The client object MUST connect to the Dapr sidecar on `127.0.0.1` and not `localhost` because some users have reported name resolution issues where `localhost` would not resolve to `127.0.0.1` or `::1`.
+
+The client object MUST abstract the protocol used to communicate to the Dapr sidecar and not offer gRPC or HTTP specific input or output.
+
+The client object MIGHT offer both gRPC or HTTP implementations.
+
+The client object SHOULD NOT implement any Actor functionality since that is handled as a separate construct.
+
+The client object MUST implement all required methods for all [core building blocks](#core-building-blocks).
+
+The client object MIGHT provide overloaded methods with reduced number of parameters for simpler method invocations.
+
+The client object SHOULD NOT implement any of the methods for the [alpha APIs](#alpha-apis)
+
+## Object Serialization
+
+The SDK MUST serialize domain objects to JSON by default, for both input and output.
+
+The SDK MUST accept `byte[]` as domain object for input and output.
+
+The SDK MIGHT accept a serializer implementation via a provided interface.
```

Comment:
Should we clarify here that Actor serialization is different from other serialization and that it will use a completely different serializer?

Reply:
Everything in Actors is separate.